libs/hdf-converters/src/sonarqube-mapper.ts

```diff
@@ -146,54 +146,50 @@ export class SonarQubeResults {
           page += 1;
         });
     }
+
     // Get code snippets for each issue
-    let requests: Promise<AxiosResponse>[] = [];
-    this.data.issues?.forEach((issue) => {
-      requests.push(
-        axios.get(`${this.sonarQubeHost}/api/sources/raw`, {
-          auth: {username: this.userToken, password: ''},
-          params: {
-            key: issue.component
-          }
-        })
-      );
-    });
-    // Wait for all requests
-    await axios.all(requests).then(
-      axios.spread((...responses) => {
-        // Extract code snippets from SonarQube
-        responses.forEach((response, index) => {
-          this.data.issues[index].snip = response.data
-            .split('\n')
-            .slice(
-              (this.data.issues[index].textRange?.startLine as number) - 3,
-              // api doesn't care if we request lines past end of file
-              (this.data.issues[index].textRange?.endLine as number) + 3
-            )
-            .join('\n');
-        });
-      })
+    await Promise.all(
+      this.data.issues?.map((issue) =>
+        axios
+          .get(`${this.sonarQubeHost}/api/sources/raw`, {
+            auth: {username: this.userToken, password: ''},
+            params: {
+              key: issue.component,
+              ...(this.branchName && {branch: this.branchName})
```

Comment:
What is the result of having logic within what is operated on by the spread operator `...`?

Reply:
Also, what is the purpose of adding in the branch name? Where does it get used?

Reply:
Understood :+1: